src/main/java/com/petrych/screenshotter/service/ScreenshotServiceImpl.java

```diff
@@ -148,35 +164,46 @@ public void delete(String urlString) throws IOException {
 		
 		UrlUtil.isUrlValid(urlString);
 		
-		String fileNameToSearchFor = findFileNameByUrl(urlString);
+		Collection<String> fileNamesToSearchFor = findFileNamesByUrl(urlString);
 		
-		if (fileNameToSearchFor.isEmpty()) {
+		if (fileNamesToSearchFor.isEmpty()) {
 			throw new FileNotFoundException();
 		} else {
-			Screenshot screenshot = ((ArrayList<Screenshot>) screenshotRepo
-					.findByNameContaining(fileNameToSearchFor)).get(0);
+			for (String fileName : fileNamesToSearchFor) {
+				Screenshot screenshot = screenshotRepo.findByFileName(fileName);
+				
+				if (screenshot != null) {
+					screenshotRepo.delete(screenshot);
+					this.deleteFile(fileName);
+					LOG.debug("Removed screenshot: {}", screenshot);
+				}
+			}
 			
-			screenshotRepo.delete(screenshot);
-			this.deleteFile(fileNameToSearchFor);
-			LOG.debug("Removed screenshot: {}", screenshot);
 		}
 	}
 	
 	@Override
-	public String findFileNameByUrl(String urlString) throws MalformedURLException {
+	public Collection<String> findFileNamesByUrl(String urlString) throws MalformedURLException {
 		
 		UrlUtil.isUrlValid(urlString);
+		String screenshotNameToSearchFor = UrlUtil.parseUrlString(urlString);
 		
-		for (String uri : findAllScreenshotUris()) {
-			String fileNameToSearchFor = ScreenshotMaker.createFileName(urlString);
+		Collection<String> screenshotFiles = new ArrayList<>();
+		Iterable<Screenshot> screenshots = findByName(screenshotNameToSearchFor);
+		
+		if (CollectionUtils.isEmpty((Collection) screenshots)) {
+			return Collections.emptyList();
+		}
+		
+		for (Screenshot screenshot : screenshots) {
+			String fileName = screenshot.getFileName();
 			
-			if (uri.endsWith(fileNameToSearchFor)) {
-				
-				return fileNameToSearchFor;
+			if (fileName != null) {
```

Comment:
Is this check done because you have screenshots in your database stored without filename? Perhaps you should consider migrating the existing screenshots first? I guess fileName will always be present for new screenshots after this PR has been merged?